### PR TITLE
#659 Updating chronicle map

### PR DESCRIPTION
**Pull request by druminski**
Version 3 of chronicle map introduces a lot of [changes](https://github.com/OpenHFT/Chronicle-Map#difference-between-chronicle-map-2-and-3). Solves #659

### CODE REVIEW

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/di/PersistentBufferExtension.java [hunk 1]
```diff
@@ -66,7 +65,13 @@ public void extend() {
         }
 
         if (config.getBooleanProperty(MESSAGES_LOCAL_STORAGE_ENABLED)) {
-            MessageRepository repository = new ChronicleMapMessageRepository(backupFilesManager.getCurrentBackupFile());
+            int backupStorageSizeInBytes = config.getIntProperty(MESSAGES_LOCAL_STORAGE_SIZE_MB) * 1024 * 1024;
```
Comment (sputnikci) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/di/PersistentBufferExtension.java]: \[Checkstyle\] INFO: '1024' is a magic number\.

Comment (sputnikci) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/di/PersistentBufferExtension.java]: \[Checkstyle\] INFO: '1024' is a magic number\.

File: hermes-frontend/build.gradle [hunk 2]
```diff
@@ -7,7 +7,7 @@ dependencies {
     compile group: 'io.undertow', name: 'undertow-core', version: '1.4.3.Final'
     compile group: 'org.apache.kafka', name: 'kafka-clients', version: '0.8.2.0'
     compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.0'
-    compile group: 'net.openhft', name: 'chronicle-map', version: '2.4.13'
+    compile group: 'net.openhft', name: 'chronicle-map', version: '3.11.0'
```
Comment (dankraw) [inline hermes-frontend/build.gradle]: we need to make sure it's backwards compatible, so we're be able to read the backup storage file written with previous version.

Comment (druminski) [inline hermes-frontend/build.gradle]: Yes, you are right, we need to test this pr very well before merge

**Comment by adamdubiel** [comment]
Further work in #711

**Pull request closed without merging**